text/0000-tool-attributes.md

```diff
@@ -216,3 +258,5 @@ Are there other tools that should be included on the whitelist (`#[test]` perhap
 
 Should we try and move some top-level attributes that are compiler-specific
 (rather than language-specific) to use `#[rustc::]`? (E.g., `crate_type`).
+
+How should the compiler expose path lints to lint plugins/lint tools?
```

Comment:
I'd assume as an initial step, We can extend [`LintPass`](http://manishearth.github.io/rust-internals-docs/rustc/lint/trait.LintPass.html) with an associated constant `&str`, which produces the tool name. While this means that clippy could accidentally produce a tool name `clippy` and `clippz` if someone makes a typo (we have like 50 `LintPass`es), this seems like a minor danger. Alternatively, we can pass the tool name via the methods on the [`Registry`](http://manishearth.github.io/rust-internals-docs/rustc_plugin/registry/struct.Registry.html), but that has the same typo issue.

As a third option, we can hide the lint registering methods behind a wrapper, which at creation time gets its tool name once, and then forwards it to the rustc data structures. That requires some more implementation effort, and can be done after the second option if it is deemed necessary.